files/default/slurm/pcluster_slurm_config_generator.py

```diff
@@ -110,31 +130,41 @@ def _get_jinja_env(template_directory):
 
 
 def _write_rendered_template_to_file(rendered_template, filename):
-    print(f"Writing contents of {filename}")
+    log.info("Writing contents of %s", filename)
     with open(filename, "w") as output_file:
         output_file.write(rendered_template)
 
 
-def main():
-    print("Running ParallelCluster Slurm Config Generator")
-    parser = argparse.ArgumentParser(description="Take in slurm configuration generator related parameters")
-    parser.add_argument(
-        "--output-directory", type=str, help="The output directory for generated slurm configs", required=True
-    )
-    parser.add_argument(
-        "--template-directory", type=str, help="The directory storing slurm config templates", required=True
-    )
-    parser.add_argument(
-        "--input-file",
-        type=str,
-        default="/opt/parallelcluster/slurm_config.json",
-        help="JSON file containing info about queues",
-    )
-    parser.add_argument(
-        "--dryrun", action="store_true", help="dryrun", required=False, default=False,
+def _setup_logger():
+    logging.basicConfig(
+        level=logging.INFO, format="%(asctime)s - [%(name)s:%(funcName)s] - %(levelname)s - %(message)s"
     )
-    args = parser.parse_args()
-    generate_slurm_config_files(args.output_directory, args.template_directory, args.input_file, args.dryrun)
+
+
+def main():
+    try:
+        _setup_logger()
+        log.info("Running ParallelCluster Slurm Config Generator")
+        parser = argparse.ArgumentParser(description="Take in slurm configuration generator related parameters")
+        parser.add_argument(
+            "--output-directory", type=str, help="The output directory for generated slurm configs", required=True
+        )
+        parser.add_argument(
+            "--template-directory", type=str, help="The directory storing slurm config templates", required=True
+        )
+        parser.add_argument(
+            "--input-file",
+            type=str,
+            default="/opt/parallelcluster/slurm_config.json",
+            help="JSON file containing info about queues",
+        )
+        parser.add_argument(
+            "--dryrun", action="store_true", help="dryrun", required=False, default=False,
+        )
+        args = parser.parse_args()
+        generate_slurm_config_files(args.output_directory, args.template_directory, args.input_file, args.dryrun)
+    except Exception as e:
+        log.exception("Failed to generate slurm configurations, exception: %s", e)
```

Comment:
I think we want the script to exit with error so that the caller can handle/fail accordingly, which means either raising or exit(1)